query.py

```diff
@@ -23,6 +23,19 @@ def _format_arg(self, arg_name, arg_val):
         '''
         return '{}:{}'.format(arg_name, arg_val)
 
+    def _query(self, http, url):
```

Comment:
I agree with your analysis...  The builtin error handling mechanism [https://en.wikipedia.org/wiki/Andrei_Alexandrescu](url) offers could of been in use.

Since I want to eventually out source this package, I will try to solve it.